interop-testing/src/main/java/io/grpc/testing/integration/TestServiceClient.java

```diff
@@ -290,16 +290,6 @@ private void runTest(TestCases testCase) throws Exception {
     @Override
     protected ManagedChannel createChannel() {
       if (!useOkHttp) {
-        InetAddress address;
-        try {
-          address = InetAddress.getByName(serverHost);
-          if (serverHostOverride != null) {
-            // Force the hostname to match the cert the server uses.
-            address = InetAddress.getByAddress(serverHostOverride, address.getAddress());
-          }
-        } catch (UnknownHostException ex) {
-          throw new RuntimeException(ex);
-        }
```

Comment:
`forAddress(InetSocketAddress)` doesn't work with proxies. This is very old code, before we had `forTarget(String)` and `forAdddress(String, port)`. Back in the day when testing TLS you had to resolve the IP yourself and then make the InetAddress manually so that it had the right name. We removed the need for that when we added `overrideAuthority()` before we went beta. Updating to the "current way" is clearer and works with CONNECT.